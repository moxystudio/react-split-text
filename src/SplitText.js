import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

export const SplitText = ({ children, className, separator, ...rest }) => {
    const words = children.split(separator);

    return words.map((word, index) => {
        const withWord = word !== '';
        const withSeparator = index !== words.length - 1;
        const shouldRender = withWord || withSeparator;

        return (
            shouldRender && <span
                key={ `word-${index}` }
                className={ classNames('react-split-text', className) }
                { ...rest }>
                { withWord && word }
                { withSeparator && separator }
            </span>
        );
    });
};

SplitText.propTypes = {
    children: PropTypes.string.isRequired,
    className: PropTypes.string,
    separator: PropTypes.string,
};

SplitText.defaultProps = {
    separator: ' ',
};

export default SplitText;
