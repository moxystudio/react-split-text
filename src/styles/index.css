.react-split-text {
    display: inline-block;
    white-space: pre-wrap;
}
