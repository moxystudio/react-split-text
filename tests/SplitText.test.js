import React from 'react';
import { render } from '@testing-library/react';
import { SplitText } from '../src/SplitText';

const renderWithProps = (props = {}) => render(<SplitText { ...props }>Hello world!</SplitText>);

describe('SplitText Component', () => {
    it('should render correctly', () => {
        const { asFragment } = renderWithProps();

        expect(asFragment()).toMatchSnapshot();
    });

    it('should render correctly with separator', () => {
        const { asFragment } = renderWithProps({ separator: '-' });

        expect(asFragment()).toMatchSnapshot();
    });
});
