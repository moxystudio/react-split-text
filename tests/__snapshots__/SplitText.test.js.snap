// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SplitText Component should render correctly 1`] = `
<DocumentFragment>
  <span
    class="react-split-text"
  >
    Hello 
  </span>
  <span
    class="react-split-text"
  >
    world!
  </span>
</DocumentFragment>
`;

exports[`SplitText Component should render correctly with separator 1`] = `
<DocumentFragment>
  <span
    class="react-split-text"
  >
    Hello world!
  </span>
</DocumentFragment>
`;
